### PR TITLE
Base64 serialization

### PR DESCRIPTION
**Pull request by JoshStoddard**
Adds header-only Base64 serialization/deserialization utilities.

### CODE REVIEW

File: include/util/Base64.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#include <vector>
+#include <cassert>
+#include <openssl/evp.h>
+
+class Base64 {
+public:
+    // static methods only
+    Base64() = delete;
+
+    static constexpr std::size_t encoded_bytes( std::size_t bytes )
```
Comment (venkat-murty) [inline include/util/Base64.h:19]: encoded_bytes and decoded_bytes are inverse functions of each other. Can we have a simple test to check that.

**Review by venkat-murty — APPROVED** [review]

File: include/util/Base64.h (hunk 1 above)
Comment (JoshStoddard) [reply include/util/Base64.h:19]: They're not quite inverse in that `encoded_bytes` always returns a multiple of 4 and `decoded_bytes` a multiple of 3.  So `decoded_bytes( encoded_bytes( 5 ) ) == 6`.

But you make a good point about adding a test.

Comment (venkat-murty) [reply include/util/Base64.h:19]: Of course you are correct.

It is strange that `decoded_bytes( encoded_bytes( 5 ) ) == 6`.

Comment (venkat-murty) [inline include/util/Base64.h:19]: Can you write a comment that these are not inverses in the code, because it seems strange.

**Review by venkat-murty — APPROVED** [review]

**Pull request merged**